hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HeapMemStoreLAB.java

```diff
@@ -107,6 +107,13 @@ public HeapMemStoreLAB(Configuration conf) {
   }
 
   /**
+   * To be used for merging multiple MSLABs
+   */
+  public void addPooledChunkQueue(BlockingQueue<PooledChunk> targetQueue) {
+    targetQueue.drainTo(pooledChunkQueue);
```

Comment:
The complexity is O(#items in targetQueue). Can not say how many items are in the queue as it depends on how many chunks we have per segment and this can vary. But I don't believe there are so many chunks that this would become a bottleneck, however this is for us to see later with stress benchmarking...

Added the comment